crates/core/src/pattern_compiler/snippet_compiler.rs

```diff
@@ -105,7 +102,7 @@ pub(crate) fn dynamic_snippet_from_source(
         parts.push(DynamicSnippetPart::String(
             source[last as usize..byte_range.start as usize].to_string(),
         ));
-        let start_pos = Position::from_byte_index(source, Some((last_pos, last)), byte_range.start);
+        let start_pos = position_from_previous(last_pos, source, last, byte_range.start);
```

Comment:
I removed the middle argument from `Position::from_byte_index()`, because it was only used for this specific call site, and made the function a bit confusing. The functionality was replaced through this `position_from_previous()` function which directly wraps `Position::from_byte_index()`.